.travis.yml

```diff
@@ -8,15 +8,13 @@ os:
 language: cpp
 
 before_install:
-  - "git fetch --unshallow"
-  - ". scripts/ci/$TRAVIS_OS_NAME/before_install.sh"
-install: ". scripts/ci/$TRAVIS_OS_NAME/install.sh"
+  - make -f Makefile.travis before_install
+install:
+  - make -f Makefile.travis install
 script:
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make debian DEBUG=1; fi
-  - ". scripts/ci/$TRAVIS_OS_NAME/script.sh"
+  - make -f Makefile.travis script
```

Comment:
No, I'm against that change. It does not help in any way to add a new makefile that calls the exact same script. It's just another indirection and source of problems/errors. It just make the builds even more complicated.

Please do not change the actual way of build the project for windows and osx. Docker for ok for the deb, but it should be added in parallel of the actual build, the same way it was added to the .travis.yml
Think that it should be easy to understand when someone needs to put its hand inside the biuld system again that it should remain easy to understand and has the same structure for all build

Reply:
Hi @raoulh 
Your code your rules :) But let's please agree for the (potential) future that you'll provide me with the guidelines and requirements before I start the work, not after I've basically finished it. No need to waste time and effort on something that isn't working as you want it to, right?
I'll be looking into your remarks today afternoon/evening.
Thanks.

Reply:
About the makefile - valid point, agreed.
Wrote it assuming the Travis-related code will grow significantly in the future so it would be nice to keep the .travis.yml file nice and clear by adding another layer of abstraction and maybe creating a makefile on a 'per OS' basis (or something in that direction).

Either way - I'll change it.